models/src/main/java/io/sphere/sdk/attributes/AttributeAccess.java

```diff
@@ -240,13 +249,13 @@ public boolean canHandle(final AttributeDefinition attributeDefinition) {
                                                              final TypeReference<Set<T>> typeReference) {
         final AttributeMapper<Set<T>> mapper = new EnumLikeSetAttributeMapperImpl<>(typeReference);
         return new AttributeAccess<>(mapper, attributeDefinition -> {
-            boolean canHandle1 = false;
-            if (SetAttributeDefinition.class.isAssignableFrom(attributeDefinition.getClass())) {
-                final SetAttributeDefinition setAttributeDefinition = (SetAttributeDefinition) attributeDefinition;
-                final SetType attributeType = setAttributeDefinition.getAttributeType();
-                canHandle1 = clazz.isAssignableFrom(attributeType.getElementType().getClass());
+            if (attributeDefinition.getAttributeType() instanceof SetType) {
+                final SetType attributeType = (SetType) attributeDefinition.getAttributeType();
+
+                return clazz.isAssignableFrom(attributeType.getElementType().getClass());
+            } else {
+                return false;
             }
-            return canHandle1;
```

Comment:
@schleichardt @yanns @gogregor @agourlay @lauraluiz I would be interested know our opinion on this refactoring. instead of the variable definition and consequent mutation of it I decided to have 2 returns. What is your opinion?